### PR TITLE
feat(auth): added implementation for updatepassword

### PR DESCRIPTION
**Pull request by banji180**
*Issue #, if available:*

*Description of changes: Added implementation for update*

*How did you test these changes?*
This is preliminary file. Test will be added later

*Documentation update required?*
No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.kt [hunk 1]
```diff
@@ -684,7 +687,56 @@ class AWSCognitoAuthPlugin : AuthPlugin<AWSCognitoAuthServiceBehavior>() {
...
+                                onSuccess: Action,
+                                onError: Consumer<AuthException>) {
+        val changePasswordRequest = ChangePasswordRequest.invoke {
+            previousPassword = oldPassword
+            proposedPassword = newPassword
+            accessToken = token
+        }
+
+        try {
+            GlobalScope.launch {
```
Comment (div5yesh) [inline aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.kt:731]: Check if user is signed in by checking the current state of AuthStateMachine using `getCurrentState`.

File: aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.kt [hunk 2]
```diff
@@ -684,7 +687,56 @@ class AWSCognitoAuthPlugin : AuthPlugin<AWSCognitoAuthServiceBehavior>() {
...
+                val cognitoClient = configureCognitoClients().cognitoIdentityProviderClient?.changePassword(changePasswordRequest)
+                onSuccess.call()
```
Comment (div5yesh) [inline aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.kt:733]: Please add unit tests.

**Review by div5yesh — CHANGES_REQUESTED** [review]

File: aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.kt [hunk 3]
```diff
@@ -684,7 +687,56 @@ class AWSCognitoAuthPlugin : AuthPlugin<AWSCognitoAuthServiceBehavior>() {
...
+            }
+        }
+        catch(e: Exception)
+        {
```
Comment (gpanshu) [inline aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.kt:737]: Formatting is off, please run `./gradlew ktlintFormat`

File: aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.kt [hunk 4]
```diff
@@ -684,7 +687,56 @@ class AWSCognitoAuthPlugin : AuthPlugin<AWSCognitoAuthServiceBehavior>() {
...
+            },
+            {
+                credentialStoreStateMachine.send(
+                    CredentialStoreEvent(CredentialStoreEvent.EventType.LoadCredentialStore())
+                )
+            }
+        )
+    }
+
+    private fun _updatePassword(token: String,
```
Comment (gpanshu) [inline aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.kt:719]: Call this val accessToken too

File: aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.kt [hunk 5]
```diff
@@ -684,7 +687,56 @@ class AWSCognitoAuthPlugin : AuthPlugin<AWSCognitoAuthServiceBehavior>() {
...
+                                onError: Consumer<AuthException>) {
+        val changePasswordRequest = ChangePasswordRequest.invoke {
+            previousPassword = oldPassword
+            proposedPassword = newPassword
+            accessToken = token
+        }
+
+        try {
+            GlobalScope.launch {
+                val cognitoClient = configureCognitoClients().cognitoIdentityProviderClient?.changePassword(changePasswordRequest)
```
Comment (gpanshu) [inline aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.kt:732]: cognitoClient is not used anymore and potentially can be deleted?

**Comment by gpanshu** [comment]
Can you update the name to conform to the PR title guideline

**Review by div5yesh — APPROVED** [review]

**Review by gpanshu — APPROVED** [review]

**Pull request merged**